processor/deltatocumulativeprocessor/internal/metrics/iter.go

```diff
@@ -0,0 +1,38 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package metrics // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/metrics"
+
+import (
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/putil/pslice"
+)
+
+func All(md pmetric.Metrics) func(func(Metric) bool) {
```

Comment:
Are we big fans of this kind of method? 🙂 

I'm pretty sure there's a reason to have this, but a function that returns a function inside a function seems overly complex at first sight.

Reply:
Yea, it's a bit complex, but it's in preparation for the golang native iterators in 1.23: https://tip.golang.org/doc/go1.23#language

Reply:
Ah ok, that makes more sense. Looking forward to cleaning up in six months!